test/integration/ingress_test.go

```diff
@@ -173,6 +173,72 @@ func TestIngressEssentials(t *testing.T) {
 	}, ingressWait, waitTick)
 }
 
+func TestGRPCIngressEssentials(t *testing.T) {
+	t.Parallel()
+	ns, cleanup := namespace(t)
+	defer cleanup()
+
+	t.Log("deploying a minimal HTTP container deployment to test Ingress routes")
+	container := generators.NewContainer("grpcbin", "moul/grpcbin", 9001)
+	deployment := generators.NewDeploymentForContainer(container)
+	deployment, err := env.Cluster().Client().AppsV1().Deployments(ns.Name).Create(ctx, deployment, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up the deployment %s", deployment.Name)
+		assert.NoError(t, env.Cluster().Client().AppsV1().Deployments(ns.Name).Delete(ctx, deployment.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeLoadBalancer)
+	// as of KTF 0.9,0 NewServiceForDeployment doesn't initialize annotations itself, need to do it outside
+	service.ObjectMeta.Annotations = map[string]string{annotations.AnnotationPrefix + annotations.ProtocolKey: "grpc"}
+	_, err = env.Cluster().Client().CoreV1().Services(ns.Name).Create(ctx, service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up the service %s", service.Name)
+		assert.NoError(t, env.Cluster().Client().CoreV1().Services(ns.Name).Delete(ctx, service.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("creating an ingress for service %s with ingress.class %s", service.Name, ingressClass)
+	kubernetesVersion, err := env.Cluster().Version()
+	require.NoError(t, err)
+	ingress := generators.NewIngressForServiceWithClusterVersion(kubernetesVersion, "/", map[string]string{
+		annotations.IngressClassKey:                             ingressClass,
+		annotations.AnnotationPrefix + annotations.ProtocolsKey: "grpc,grpcs",
+		annotations.AnnotationPrefix + annotations.StripPathKey: "false",
+	}, service)
+	require.NoError(t, clusters.DeployIngress(ctx, env.Cluster(), ns.Name, ingress))
+
+	defer func() {
+		t.Log("cleaning up Ingress resource")
+		if err := clusters.DeleteIngress(ctx, env.Cluster(), ns.Name, ingress); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
```

Comment:
This is done elsewhere in this patch, but generally we use `asserts` for cleanup instead of requires, the idea being that if one cleanup fails we still want to at least try and perform the others:
```suggestion
				assert.NoError(t, err)
```